cmd/tink-worker/client/tink/tink_test.go

```diff
@@ -0,0 +1,158 @@
+ZSBDbzCBmzAQBgcqhkjOPQIBBgUrgQQAIwOBhgAEAL71xJa5btFlW1BUaxrLAbsM
+1tgi0UTB6w4CHm+6iwyuP0BZD9eMwSjMzhZ70lmlah0Z5Z8oHFHEFOIYZnIG4O/r
+ATnzswVeVIPuX8uZ6ApsKq5q9uk9ByJIRfLhNqmrpLNSGbdcNEIoI27DIwiQdeT8
+U6WYuWZB8BKBNre2Q/OiAcPNozUwMzAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAww
+CgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAKBggqhkjOPQQDBAOBjAAwgYgCQgE1
+AkrK0PoP1Mvia0jMGYBDwGzBCLjqPpPFuTE0pVJctog/ZXBRUDlSz9BN3PbCFJZs
+F+/RUdtxQsgcBeBNrKk2qwJCAVxHlTzNK+hv7Xr3MXC39TopWJuNEj/1B37FZ8HG
+0qv7ie1O3lcrMXd9evsIp9KLqSgUCyxmN2SS6LKBjaAwzmJd
+-----END CERTIFICATE-----`
+
+type fakeServer struct {
+	workflow.WorkflowServiceServer
+	data map[string]string
+	err  error
+}
+
+const bufSize = 1024 * 1024
+
+func startWorkflowServerAndConnectClient(t *testing.T, name string, server *grpc.Server) workflow.WorkflowServiceClient {
+	t.Helper()
+
+	ctx := context.Background()
+	listener := bufconn.Listen(bufSize)
+	go func() {
+		t.Helper()
+
+		if err := server.Server().Serve(listener); err != nil {
+			t.Error(fmt.Errorf("%s.Serve exited with error: %w", name, err))
+		}
+	}()
+
+	dialer := func(ctx context.Context, _ string) (net.Conn, error) {
+		return listener.DialContext(ctx)
+	}
+
+	conn, err := ggrpc.DialContext(ctx, "bufnet", ggrpc.WithContextDialer(dialer), ggrpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial %s: %v", name, err)
+	}
+
+	client := workflow.NewWorkflowServiceClient(conn)
+	return client
+}
+
+func TestObtainTinkServerCreds(t *testing.T) {
+	certTests := []struct {
+		testName         string
+		url              string
+		expectedErrorMsg string
+	}{
+		{"certURL arg validation", "", "certURL cannot be empty"},
+		{"ensure we get a response when fetching a cert", "something", "unable to fetch cert from something"},
+	}
+	for _, tt := range certTests {
+		t.Run(tt.testName, func(t *testing.T) {
+			_, err := ObtainTinkServerCreds(tt.url)
+			if err == nil {
+				t.Error("Error expected but none found")
+			} else if fmt.Sprint(err) != tt.expectedErrorMsg {
+				t.Errorf("Error should be %v, got %v", tt.expectedErrorMsg, err)
+			}
+		})
+	}
+
+	t.Run("ensure an empty cert response is rejected", func(t *testing.T) {
+		emptyCert := ""
+		svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, emptyCert)
+		}))
+		defer svr.Close()
+
+		expectedErrorMsg := "unable to parse the cert data"
+		_, err := ObtainTinkServerCreds(svr.URL)
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+
+	t.Run("using with a self-signed cert", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, selfSignedCert)
+		}))
+		defer server.Close()
+
+		_, err := ObtainTinkServerCreds(server.URL)
+		if err != nil {
+			t.Errorf("Error obtaining tink server creds: %v", err)
+		}
+	})
```

Comment:
you might want to consider moving these to a separate test or finding a way to add them to the already existing table tests. Typically a single table test is used per test function ([ref](https://go.dev/doc/code#Testing)).